### PR TITLE
Feature/error handling

### LINKED ISSUE #18: Cannot access to sentry-youtrack configuration page after configuring a wrong youtrack URL
After configuring a wrong youtrack URL for sentry-youtrack plugin, I cannot access to sentry-youtrack configuration page to update that URL.
When selecting the Youtrack tab of that project, I got below error:
```
 Oops! Something went wrong
It looks like you've hit an issue in our server API. Don't worry though! We use Sentry to monitor Sentry and it's likely we're already looking into this!
If you're daring, you may want to try the following:
Give it a few seconds and reload the page.
```
In the log:
```
sentry[14332] ERROR django.request base.handle_uncaught_exception:224 Internal Server Error: /team/project/settings/plugins/youtrack/
    Traceback (most recent call last):
      File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/../django/core/handlers/base.py", line 112, in get_response
        response = wrapped_callback(request, *callback_args, **callback_kwargs)
      File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/../django/views/generic/base.py", line 69, in view
        return self.dispatch(request, *args, **kwargs)
      File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/../django/utils/decorators.py", line 29, in _wrapper
        return bound_func(*args, **kwargs)
      File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/../django/utils/decorators.py", line 99, in _wrapped_view
        response = view_func(request, *args, **kwargs)
      File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/../django/utils/decorators.py", line 25, in bound_func
        return func(self, *args2, **kwargs2)
      File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/web/frontend/base.py", line 172, in dispatch
        return self.handle(request, *args, **kwargs)
      File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/web/frontend/project_plugin_configure.py", line 22, in handle
        view = plugin.configure(request, project=project)
      File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/plugins/base/v1.py", line 472, in configure
        return default_plugin_config(self, project, request)
      File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/plugins/base/configuration.py", line 37, in default_plugin_config
        prefix=plugin_key,
      File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/../sentry_youtrack/forms.py", line 194, in __init__
        for project in client.get_projects():
      File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/../sentry_youtrack/youtrack.py", line 157, in get_projects
        for project in BeautifulSoup(response.text, 'xml').projects:
    TypeError: 'NoneType' object is not iterable
```
I tried to disable sentry-youtrack plugin and enable it again but it did not help.

sentry version: 8.4.0
sentry-youtrack plugin version: 0.3.1
YouTrack version: youtrack-6.5.17105
beautifulsoup4: 4.5.1

thanks & best regards.

### PR DESCRIPTION
**Pull request by bogdal**
Fixes #18

### CODE REVIEW

File: sentry_youtrack/forms.py [hunk 1]
```diff
@@ -168,43 +169,59 @@ class YouTrackConfigurationForm(forms.Form):
...
+        else:
+            names = [field['name'] for field in fields]
+            return zip(names, names)
+        return []
+
+    def get_project_field_choices(self, client, project):
+        choices = [(' ', u"- Choose project -")]
+        try:
+            projects = list(client.get_projects())
+        except HTTPError:
```
Comment (quantonganh) [inline sentry_youtrack/forms.py]: I think we should also handle the `TypeError` exception here:

```
          File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/../sentry_youtrack/forms.py", line 186, in __init__
            client, initial.get('project'))
          File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/../sentry_youtrack/forms.py", line 220, in get_project_field_choices
            projects = list(client.get_projects())
          File "/usr/local/sentry/local/lib/python2.7/site-packages/sentry/../sentry_youtrack/youtrack.py", line 157, in get_projects
            for project in BeautifulSoup(response.text, 'xml').projects:
        TypeError: 'NoneType' object is not iterable
```

**Comment by quantonganh** [comment]
But the problem is we should have to validate the YouTrack URL before saving.

#18 is not fixed completely. You can try something like this:

- YouTrack Instance URL: enter an URL to specific issue (https://yt.domain.com/issue/xx-123) instead of the base URL (https://yt.domain.com)
- Username:
- Password: try to put an incorrect password
- Save Changes

you will get the error "Invalid project: 'None'".
But you cannot update the URL or choose a Linked Project (as it's empty).

File: sentry_youtrack/forms.py [hunk 2]
```diff
@@ -132,6 +132,7 @@ class YouTrackConfigurationForm(forms.Form):
         'client': _("Unable to connect to YouTrack."),
         'invalid_ssl': _("SSL certificate  verification failed."),
         'invalid_password': _('Invalid username or password.'),
+        'invalid_project': _('Invalid project: \'%s\''),
```
Comment (quantonganh) [inline sentry_youtrack/forms.py]: Should it be `invalid_url`?

File: sentry_youtrack/forms.py (hunk 1 above)
Comment (bogdal) [inline sentry_youtrack/forms.py]: We want to handle only the case when the current project is no longer available. `TypeError` indicates problems with API.

File: sentry_youtrack/forms.py (hunk 2 above)
Comment (bogdal) [inline sentry_youtrack/forms.py]: We already have `client` message for such problems.

**Comment by bogdal** [comment]
The problem is `YouTrackClient` that accepts wrong url instead of throwing an exception.

File: sentry_youtrack/forms.py (hunk 1 above)
Comment (quantonganh) [inline sentry_youtrack/forms.py]: > We want to handle only the case when the current project is no longer available.

ah, ok.

> `TypeError` indicates problems with API.

Yes, I know.
That is the case when user enter a wrong URL (but still can login) as I mentioned below.

File: sentry_youtrack/forms.py (hunk 2 above)
Comment (quantonganh) [inline sentry_youtrack/forms.py]: No.
`client` message is used when we cannot connect to YouTrack.
Here we can but cannot get any project as the URL is wrong.

**Pull request merged**